graph/graph.go

```diff
@@ -26,9 +26,10 @@ import (
 
 // A Graph is a store for versioned filesystem images and the relationship between them.
 type Graph struct {
-	Root    string
-	idIndex *truncindex.TruncIndex
-	driver  graphdriver.Driver
+	Root        string
+	idIndex     *truncindex.TruncIndex
+	driver      graphdriver.Driver
+	pullingImgs map[string][]string
```

Comment:
Could this be generalized to "marked" or "retained" something like that? I can envision other cases where hanging on to an image during an operation may be desirable.